realsense/plugin.cpp

```diff
@@ -302,6 +305,10 @@ class realsense : public plugin {
         }
     }
 
+
+	std::optional<ullong> _m_first_imu_time;
+	std::optional<time_point> _m_first_real_time;
```

Comment:
To check for first use and assignment?

Reply:
These are unset when the object is constructed, and set during the first execution of `callback`.

Using a `std::optional<...>` is better than using a sentinel value, such as 0, because you don't have to assume things about the domain of the variable.